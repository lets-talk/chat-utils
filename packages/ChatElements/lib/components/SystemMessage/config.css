:root {
  --systemMessage-font-size: var(--font-size-small);
  --systemMessage-background: var(--palette-orange-900);
  --systemMessage-color: var(--color-white);
}
