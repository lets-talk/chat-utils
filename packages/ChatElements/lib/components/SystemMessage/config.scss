@import "../../common/styles/variables";

$systemMessage-font-size: $font-size-small;
$systemMessage-background: $color-white;
$systemMessage-color: rgb(113, 131, 150);
$systemMessage-boxShadow: 1px 1px 1px 1px rgba(0,0,0,.2);
