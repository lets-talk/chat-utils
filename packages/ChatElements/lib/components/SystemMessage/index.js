import React from 'react';
import PropTypes from 'prop-types';

import './index.css';

const classNames = require('classnames');

const SystemMessage = (props) =>
  (
    <div className={classNames('letstalk-container-smsg', props.className)}>
      <div
        className="letstalk-smsg"
      >
        <div className="letstalk-smsg-text">
          {props.text}
        </div>
      </div>
    </div>
  );

SystemMessage.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
};
export default SystemMessage;
