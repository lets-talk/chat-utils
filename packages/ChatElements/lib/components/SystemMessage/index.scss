@import './config.scss';

.letstalk-container-smsg {
    display: flex;
    align-items: center;
    justify-content: center;
}

.letstalk-smsg {
    position: relative;
    background: $systemMessage-background;
    border-radius: 10px;
    box-shadow: $systemMessage-boxShadow;
    display: flex;
    flex-direction: column;
    margin: 5px 0px;
    padding: 6px 9px 8px 9px;
    float: left;
    max-width: 70%;
    align-items: center;
    justify-content: center;
}

.letstalk-smsg-text {
    text-align: center;
    display: inline-block;
    font-size: $systemMessage-font-size;
    color: $systemMessage-color;
}
