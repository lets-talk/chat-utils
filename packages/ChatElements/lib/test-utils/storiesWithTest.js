import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
