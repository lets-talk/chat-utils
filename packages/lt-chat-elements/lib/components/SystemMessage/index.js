import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { textColor } from '../../utils/style';

const StyledContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

  div {
    position: relative;
    display: flex;
    flex-direction: column;
    margin: 5px 0px;
    padding: 6px 9px 8px 9px;
    float: left;
    max-width: 70%;
    align-items: center;
    justify-content: center;

    div {
      text-align: center;
      line-height: 1;
      display: inline-block;

      color: ${(props) => textColor(props.theme, 'light', 'disabled')};
      font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
      line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
      font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};

      span {
        display: inline-block;
        position: relative;
      }

      span:before,
      span:after {
        content: "";
        position: absolute;
        height: 10px;
        border-bottom: 1px solid ${(props) => textColor(props.theme, 'light', 'disabled')};
        top: 0;
        width: 100%;
      }
      span:before {
        right: 100%;
        margin-right: 15px;
      }
      span:after {
        left: 100%;
        margin-left: 15px;
      }
    }
  }
`;

const SystemMessage = (props) => {
  const { message } = props;
  return (
    <StyledContainer className={props.className}>
      <div>
        <div>
          <span>{message.text}</span>
        </div>
      </div>
    </StyledContainer>
  );
};

SystemMessage.propTypes = {
  className: PropTypes.string,
  message: PropTypes.shape({
    text: PropTypes.string,
  }),
};

SystemMessage.displayName = 'SystemMessage';

export default SystemMessage;
