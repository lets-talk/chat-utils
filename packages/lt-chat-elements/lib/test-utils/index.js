// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
