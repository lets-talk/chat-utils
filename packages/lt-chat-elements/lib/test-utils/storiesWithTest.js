import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
