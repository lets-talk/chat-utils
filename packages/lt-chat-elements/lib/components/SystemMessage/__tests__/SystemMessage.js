import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import SystemMessage from '../index';
import { renderWithTheme } from '../../../test-utils';

describe('SystemMessage component', () => {
  it('should render without issues', () => {
    const message = { text: 'Someone jointed the conversation' };
    const component = shallow(<SystemMessage message={message} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('Styled: Should Render expected Styles', () => {
    const message = { text: 'Someone jointed the conversation' };
    const tree = renderWithTheme(<SystemMessage message={message} />).toJSON();
    expect(tree).toMatchSnapshot();
    // expect(tree).toHaveStyleRule('background-color', 'rgba(94,124,139,0.1)');
  });
});
