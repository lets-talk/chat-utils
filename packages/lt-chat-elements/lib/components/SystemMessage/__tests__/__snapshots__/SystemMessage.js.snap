// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemMessage component Styled: Should Render expected Styles 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 div {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 5px 0px;
  padding: 6px 9px 8px 9px;
  float: left;
  max-width: 70%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 div div {
  text-align: center;
  line-height: 1;
  display: inline-block;
  color: rgba(0,0,0,0.38);
  font-size: calc(12 * 1px);
  line-height: calc(16 * 1px);
  font-weight: 500;
}

.c0 div div span {
  display: inline-block;
  position: relative;
}

.c0 div div span:before,
.c0 div div span:after {
  content: "";
  position: absolute;
  height: 10px;
  border-bottom: 1px solid rgba(0,0,0,0.38);
  top: 0;
  width: 100%;
}

.c0 div div span:before {
  right: 100%;
  margin-right: 15px;
}

.c0 div div span:after {
  left: 100%;
  margin-left: 15px;
}

<div
  className="c0"
>
  <div>
    <div>
      <span>
        Someone jointed the conversation
      </span>
    </div>
  </div>
</div>
`;

exports[`SystemMessage component should render without issues 1`] = `
<styled.div>
  <div>
    <div>
      <span>
        Someone jointed the conversation
      </span>
    </div>
  </div>
</styled.div>
`;
